### PR TITLE
feat: project non-queryable metadata

### PR DESCRIPTION
**Pull request by mhagel**
## Description

Closes: regen-network/regen-registry#1422

- Splits `ProjectMetadataLD` (anchored) from `ProjectPageMetadataLD` (unanchored)
- Refactors some `ProjectDetails` and `ProjectTopSection` code for readability, modularity
- Project metadata queries all return compacted JSON-LD, which improves readability, eg instead of using `metadata['schema:image']['@value']` we can just use `metadata['schema:image']`

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. https://deploy-preview-1701--regen-registry.netlify.app/project/C45-020 is a new staging/redwood project using updated split metadata.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts [hunk 1]
```diff
@@ -1,39 +1,61 @@
...
   'regen:projectQuote': ProjectQuote;
   'regen:landStewardStoryTitle': string;
+  'schema:description'?: string;
+
+  // legacy? - may not exist in standards repo
+  'regen:boundaries': TypeValue;
+  'regen:creditClass': CreditClass;
+  'regen:landManagementActions': LandManagementActions;
+  'regen:glanceText': any;
 }
```
Comment (mhagel) [inline web-registry/src/lib/db/types/json-ld/project-metadata.ts:53]: @blushi @S4mmyb @clevinson 

After talking with @wgwz, we think  `ProjectPageMetadataLD` is useful naming for what we were calling "non-queryable" metadata. This maintains some context across repos. Queryable metadata would remain `ProjectMetadataLD`. WDYT?

Also, the fields you see in this interface represent our best understanding of where we landed in our last meeting together. Do you agree these should all be items that can be edited without signing a ledger TX? 

Let's discuss here if you have any evolving thoughts on any of these. Otherwise, @wgwz and I will continue in this direction. Thanks!

Comment (blushi) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:53]: I believe that looks good.
What about `regen:projectOperator` for CFC projects? on project pages, this data appears under "project developer" label

Comment (mhagel) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:53]: I wondered about that too. I think it should remain in the "signed" data. Even though it is similar to the roles, it seems very coupled to the CFC project.

Comment (blushi) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:53]: So after discussing this with @S4mmyb, we should just use project developer in place of project operator in this case.
Also, regarding roles in general, it's preferable to just store them through the existing references in the project table (`developer_id`, `originator_id`, etc.) and create `party`s for them if they don't exist yet (just like it's already done for carbonplus grasslands and kasigau/mai-ndombe projects I believe). Planning ahead for the user profiles stories where project stakeholders could edit their profile and that would be reflected on all the projects they are related to.

Comment (mhagel) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:53]: Ok, so we would remove `regen:projectOperator` from the[ C02 project shape](https://github.com/regen-network/regen-registry-standards/blob/7e32d372161c12b3145fb674e1e0a6613b670119/jsonld/projects/C02-project.jsonld#L43), and reassign the data for those existing projects to the `regen:projectDeveloper`, correct?

Comment (wgwz) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:53]: >Also, regarding roles in general, it's preferable to just store them through the existing references in the project table (developer_id, originator_id, etc.) and create partys for them if they don't exist yet (just like it's already done for carbonplus grasslands and kasigau/mai-ndombe projects I believe). Planning ahead for the user profiles stories where project stakeholders could edit their profile and that would be reflected on all the projects they are related to.

I might be misunderstanding and so it's probably worth talking through it some more. But I don't see how we would be able to do this now. 

Here's the info available to us now for `projectOperator` in the project metadata (to be renamed `projectDeveloper`):
Screen Shot 2022-12-15 at 3 14 35 PM

Here's what we require in the party table (i.e. a wallet address):
Screen Shot 2022-12-15 at 3 16 25 PM

Wouldn't we need to gather up a wallet address to assign for this purpose, if we are going to make use of these references now?

It might be better to wait until we have the account creation process (the login system) in place before we start populating parties

Comment (blushi) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts]: I understand your point but wallet_id is not required here.
If the current data structure is gonna stay the same with the login system, mainly parties being referenced in the project table then I believe that's fine to start populating parties.

Comment (wgwz) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts]: i’ll make some time to review the changes i have in the login PR, i think i just need some time to see how the pieces fit together. i am just thinking about how we were trying to do the whole thing where an account has a corresponding party that has a corresponding wallet (separate tables). and if we put this data in now, for example, how would we know which party needs to be used during the account creation process? i think the entries we put in now, may just end stale data? that would be difficult to manually tie in to the login system?

Comment (mhagel) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts]: Thinking about the project create flow. 

Our UI adds a party to the postgres DB when we add a Role (project developer). So, if we add more CFC projects in the future, we will have to make sure the Project Developer is added in that Roles step. If we consider this a required field for CFC, we will want to add some validation to enforce it. 

But maybe that is not necessary.

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts [hunk 2]
```diff
@@ -1,37 +1,44 @@
 import { TypeValue, UrlList, UrlType } from 'lib/rdf/types';
```
Comment (mhagel) [inline web-registry/src/lib/db/types/json-ld/project-metadata.ts:1]: The new interface `ProjectPageMetadataLD` on this page pulls fields from `ProjectMetadataLD`.

These changes guide where display items come from in `ProjectDetails` and its nested components.

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx [hunk 3]
```diff
@@ -284,19 +289,20 @@ function ProjectTopSection({
         </Grid>
         <Grid item xs={12} md={4} sx={{ pt: { xs: 10, sm: 'inherit' } }}>
           <ProjectTopCard
-            projectAdmin={getDisplayAdmin(data?.admin || onChainProject?.admin)}
```
Comment (mhagel) [inline web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:287]: I don't think `data?.admin` was ever getting used here, maybe doesn't exist.

File: web-registry/src/lib/db/types/json-ld/cfc-project-metadata.ts [hunk 4]
```diff
@@ -9,16 +9,10 @@ export interface CFCProjectMetadataLD extends ProjectMetadataLD {
   'regen:cfcProjectPage': TypeValue;
   'regen:offsetProtocol': OffsetProtocol;
   'regen:projectDesignDocument': TypeValue;
-  'regen:projectOperator'?: ProjectOperator;
```
Comment (mhagel) [inline web-registry/src/lib/db/types/json-ld/cfc-project-metadata.ts:12]: `regen:projectOperator` being removed from CFC, in favor of existing  `regen:projectDeveloper`

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts (hunk 1 above)
Comment (wgwz) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts]: Here are some thoughts I've had thinking through the current database model in comparison to the database model in the login system PR. Some of these notes are tangential to our discussion here and we can consider these separately. Please correct me if I make any incorrect statements!

In the current model:
- Each `party` can have many `project` references.
- But each `project` can only have one developer, one originator, etc.
- A single `party` can represent one or more of these roles:
    - a developer, originator, broker, land owner, registry, reseller and steward. do we still need all of these roles?
- The `project` table has a ref to `wallet.id` table, do we still need this ref?
- The `project` table has a ref to `user.id` table called `creator_id`, do we still need this ref?

In the login model:
- A given `party` represents a profile that is tied to a unique `wallet` address.
- A `party` represents a particular `wallet` address and is tied to an `account`.

A mutual goal in both of these models is that a `party` represents a "profile". So indeed, in this way being able to link for example, user profiles, to a particular project can be achieved. But thinking towards when we do implement the login system, we may have to re-think the UI for attaching a particular wallet as a project developer (for example).

TL;DR I think the migration to login model will be fairly tricky. And we should try to do as much clean up as we can in the database to make the model easier to think about.

Comment (mhagel) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts]: >Q: In the edit project workflow, do we have a way to edit who the project developer is?

Yes. But the code may need some updating.

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts [hunk 5]
```diff
@@ -1,37 +1,44 @@
...
-  'schema:description'?: string;
   'schema:location': any;
   'regen:projectType': string;
   'regen:projectActivity': ProjectActivity;
-  'regen:projectDeveloper': ProjectStakeholder;
-  'regen:landSteward': ProjectStakeholder;
-  'regen:landOwner': ProjectStakeholder;
-  'regen:projectOriginator': ProjectStakeholder;
   'regen:offsetGenerationMethod': string;
   // TODO: not sure why regen:offsetGenerationMethod isn't compacting as expected so added this:
```
Comment (wgwz) [inline web-registry/src/lib/db/types/json-ld/project-metadata.ts:12]: is this still true? i kind of remember fixing this previously, or something similar to it

Comment (wgwz) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:12]: i just made a connection between this and another issue i opened in registry standards that we need to fix: https://github.com/regen-network/regen-registry-standards/issues/35

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts [hunk 6]
```diff
@@ -1,37 +1,44 @@
 import { TypeValue, UrlList, UrlType } from 'lib/rdf/types';
 
+/** Anchored metadata AKA "Additional Info" - Editable only with a signed Ledger TX. */
 export interface ProjectMetadataLD {
```
Comment (wgwz) [inline web-registry/src/lib/db/types/json-ld/project-metadata.ts:4]: one question i have is where is the `vcsProjectId` for the C01 projects? i.e. https://github.com/regen-network/regen-registry-standards/blob/main/jsonld/projects/C01-verified-carbon-standard-project.jsonld#L11

i'm wondering whether i should put this in the `C01-Project` or `C01-Project-Page` schema, or do i need to think about it differently?

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts [hunk 7]
```diff
@@ -1,37 +1,44 @@
...
-  'regen:landManagementActions': LandManagementActions;
+}
+
+/** Un-anchored metadata from our DB. This is editable without a Ledger TX. */
+export interface ProjectPageMetadataLD {
+  '@context': Context;
+  '@type': string; // regen:C01-Project-Page
+  'schema:image': UrlType;
+  'schema:creditText': string;
   'regen:galleryPhotos': UrlList;
```
Comment (wgwz) [inline web-registry/src/lib/db/types/json-ld/project-metadata.ts:25]: note: some C01 in the ops repo don't have `galleryPhotos` or `previewPhoto`. do we need to mark those as optional in this interface? i.e. https://github.com/regen-network/regen-registry-standards/blob/main/ops/C01/project-metadata/C01-ProjectId1218-metadata.jsonld

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts [hunk 8]
```diff
@@ -1,37 +1,44 @@
...
   'regen:landStewardPhoto': UrlType;
   'regen:projectQuote': ProjectQuote;
   'regen:landStewardStoryTitle': string;
+  'schema:description'?: string;
+  'regen:boundaries': TypeValue;
+  'regen:creditClass': CreditClass;
+  'regen:landManagementActions': LandManagementActions;
 }
 
 interface Context {
```
Comment (wgwz) [inline web-registry/src/lib/db/types/json-ld/project-metadata.ts:44]: nit/tangent/optional: we could technically be more strict with the `Context` interface:

```
interface Context {
  schema: 'http://schema.org/';
  regen: 'http://regen.network/';
  qudt: 'http://qudt.org/schema/qudt/';
  unit: 'http://qudt.org/vocab/unit/';
  xsd: 'http://www.w3.org/2001/XMLSchema#';
}
```

Comment (wgwz) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:44]: but there are some fields in the context that may need to be marked as optional. i.e. not all metadata needs the `xsd` schema for example.

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts (hunk 6 above)
Comment (wgwz) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:4]: as i'm working on changes in the registry standards, it seems like it might be helpful to have a project identifier (i.e. `vcsProjectId`) in both the `C01-Project` and the `C01-Project-Page` metadata

Comment (mhagel) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:4]: agreed

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts (hunk 7 above)
Comment (mhagel) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:25]: Good point. We are using `Partial<T>` to make each field optional in code right now. I am going to keep thinking about whether it would be better to explicitly set optional fields here.

**Review by flagrede — COMMENTED** [review]
Looks great.
We'll probably have some conflicts with #1689 indeed.
Left just one minor comment on metadata hooks, will give it a second review later.

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts (hunk 1 above)
Comment (blushi) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts]: > - The `project` table has a ref to `wallet.id` table, do we still need this ref?
> - The `project` table has a ref to `user.id` table called `creator_id`, do we still need this ref?

I think we might want to keep the former moving forward.

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts (hunk 7 above)
Comment (blushi) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:25]: I think it would best to make it explicit since we do know which fields are required and which ones are optional
for instance, we always have some `@context` and `@type`
using `Partial<ProjectPageMetadataLD>` allows those fields to not be there...

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts [hunk 9]
```diff
@@ -1,60 +1,63 @@
...
+/** Un-anchored metadata from our DB. This is editable without a Ledger TX. */
+export interface ProjectPageMetadataLD {
+  '@context': Context;
+  '@type': string; // regen:C01-Project-Page
+  'schema:image': UrlType;
+  'schema:creditText': string;
   'regen:galleryPhotos': UrlList;
   'regen:previewPhoto': UrlType;
   'regen:videoURL': UrlType;
+  'regen:projectDeveloper': ProjectStakeholder;
```
Comment (blushi) [inline web-registry/src/lib/db/types/json-ld/project-metadata.ts:28]: as stated in https://github.com/regen-network/regen-web/pull/1701#discussion_r1049989145, we might just want to remove those roles from the json ld metadata and have them as project references

File: web-registry/src/components/organisms/ProjectTopSection/hooks/useProject.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+import { CreditClass, CreditClassVersion, Maybe } from 'generated/graphql';
+
+type UseProjectResponse = {
+  creditClass?: Maybe<CreditClass>;
+  creditClassVersion?: Maybe<CreditClassVersion>;
+  sdgIris?: any[];
+  offsetGenerationMethod?: string;
+};
+
+export const useProject = (project?: any): UseProjectResponse => {
```
[the remark this answers is not in the document]
Comment (mhagel) [reply web-registry/src/components/organisms/ProjectTopSection/hooks/useProject.ts]: done

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts (hunk 9 above)
Comment (mhagel) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts]: As of latest convo with @wgwz, leaving these in anchored metadata

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts (hunk 7 above)
Comment (mhagel) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts]: Using explicit optionals now.

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts (hunk 6 above)
Comment (mhagel) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts:4]: added these as optionals in `ProjectPageMetadataLD`

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts [hunk 11]
```diff
@@ -1,60 +1,70 @@
...
+  'regen:landStewardStory'?: string;
+  'regen:landStewardStoryTitle'?: string;
+  'regen:landStewardPhoto'?: UrlType;
+  'regen:projectQuote'?: ProjectQuote;
+  'schema:description'?: string;
+  'regen:boundaries'?: TypeValue;
+  'regen:creditClass'?: CreditClass;
+  'regen:landManagementActions'?: LandManagementActions;
+
+  // Optional reference IDs:
```
Comment (mhagel) [inline web-registry/src/lib/db/types/json-ld/project-metadata.ts:45]: @blushi Discussing with @wgwz , these may be helpful knowlege graph reference IDs. If we decide to keep them here in `ProjectPageMetadataLD`, we will need to make sure they get updated at the same time as the anchored ones.

File: web-registry/src/components/organisms/ProjectTopSection/hooks/useAnchoredMetadata.ts [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+import { getAreaUnit, qudtUnit } from 'lib/rdf';
+
+type UseAnchoredMetadataResponse = {
+  projectName?: string;
+  area?: string;
+  areaUnit?: string;
+  placeName?: string;
+};
+
+export const useAnchoredMetadata = (
```
[the remark this answers is not in the document]
Comment (mhagel) [reply web-registry/src/components/organisms/ProjectTopSection/hooks/useAnchoredMetadata.ts:13]: I was wondering about this. I thought maybe this would be more performant because it only updates if `anchoredMetadata` updates.

File: web-registry/src/pages/ChooseCreditClassPage/ChooseCreditClassPage.tsx [hunk 13]
```diff
@@ -1,12 +1,102 @@
 import React from 'react';
```
Comment (mhagel) [inline web-registry/src/pages/ChooseCreditClassPage/ChooseCreditClassPage.tsx:1]: This file used to be `ChooseCreditClass`. Since we are deleting `ChooseCreditClassLegacy`, I renamed this to `ChooseCreditClassPage` because that was just a wrapper for this file.

Comment (mhagel) [reply web-registry/src/pages/ChooseCreditClassPage/ChooseCreditClassPage.tsx:1]: This may look as if `ChooseCreditClass` was deleted, but really it was just renamed.

File: web-registry/src/components/organisms/ProjectTopSection/hooks/useAnchoredMetadata.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (mhagel) [reply web-registry/src/components/organisms/ProjectTopSection/hooks/useAnchoredMetadata.ts:11]: got rid of useState and useEffect

File: web-registry/src/components/organisms/ProjectTopSection/hooks/useProjectPageMetadata.ts [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+  videoURL: undefined,
+  glanceText: undefined,
+  primaryDescription: undefined,
+  quote: undefined,
+  landStewardPhoto: undefined,
+  landStewardStoryTitle: undefined,
+  landStewardStory: undefined,
+};
+
+export const useProjectPageMetadata = (
```
[the remark this answers is not in the document]
Comment (mhagel) [reply web-registry/src/components/organisms/ProjectTopSection/hooks/useProjectPageMetadata.ts:25]: done

File: web-registry/src/components/organisms/ProjectTopSection/hooks/useProject.ts [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+  const [values, setValues] = useState<UseProjectResponse>({});
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (mhagel) [reply web-registry/src/components/organisms/ProjectTopSection/hooks/useProject.ts]: done. and renamed to `useProjectDetails`

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts [hunk 16]
```diff
@@ -1,60 +1,63 @@
...
+  'regen:landOwner': ProjectStakeholder;
+  'regen:projectOriginator': ProjectStakeholder;
   'regen:landStory': string;
   'regen:landStewardStory': string;
   'regen:landStewardPhoto': UrlType;
   'regen:projectQuote': ProjectQuote;
   'regen:landStewardStoryTitle': string;
+  'schema:description'?: string;
+  'regen:boundaries': TypeValue;
+  'regen:creditClass': CreditClass;
```
[the remark this answers is not in the document]
Comment (mhagel) [reply web-registry/src/lib/db/types/json-ld/project-metadata.ts]: Done

**Comment by mhagel** [comment]
https://deploy-preview-1701--regen-registry.netlify.app/project/C45-020 is an example project using data from 

https://github.com/regen-network/regen-registry-standards/blob/kyle/1420-project-metadata-separation/ops/C02/project-metadata/C02-project-page-003.jsonld
and
https://github.com/regen-network/regen-registry-standards/blob/kyle/1420-project-metadata-separation/ops/C02/project-metadata/C02-project-003.jsonld

TODO: display project stakeholders (`regen:projectOperator` in this case)

**Comment by blushi** [comment]
@flagrede @wgwz I would appreciate another eye on this

File: web-registry/src/lib/db/types/json-ld/project-metadata.ts [hunk 17]
```diff
@@ -1,69 +1,82 @@
...
-  'regen:landOwner': ProjectStakeholder;
-  'regen:projectOriginator': ProjectStakeholder;
+  'regen:projectActivity': NameUrl;
   'regen:offsetGenerationMethod': string;
-  // TODO: not sure why regen:offsetGenerationMethod isn't compacting as expected so added this:
+  // regen:offsetGenerationMethod doesn't get compacted for project metadata because
+  // in COMPACTED_CONTEXT, we specify: 'regen:offsetGenerationMethod': { '@container': '@list' }
+  // while in project metadata, it's a string, not a list
+  // this context is used in credit classes metadata, we should probably update the credit classes metadata
+  // to use offsetGenerationMethods (plural!) since it can be a list
```
Comment (blushi) [inline web-registry/src/lib/db/types/json-ld/project-metadata.ts:16]: @wgwz I spotted some inconsistencies in our projects/credit classes metadata which is causing some issue on the front-end when trying to compact the metadata
we always use `regen:offsetGenerationMethod` singular even when there can be multiple in credit class metadata (C01 has "Emission Reductions" and "Carbon Removals" for instance)

cc/ @S4mmyb

File: web-registry/src/components/molecules/ProjectPageMetadata/ProjectPageMetadata.tsx [hunk 18]
```diff
@@ -5,13 +5,13 @@ import { ExpandButton } from 'web-components/lib/components/buttons/ExpandButton
 import { Title } from 'web-components/lib/components/typography';
 import { formatDate } from 'web-components/lib/utils/format';
 
-import { ProjectMetadataIntersectionLD } from 'lib/db/types/json-ld';
+import { AnchoredProjectMetadataIntersectionLD } from 'lib/db/types/json-ld';
 
 import { ArrowLink } from '../../atoms/MetadataArrowLink';
 import { MetaDetail } from '../MetaDetail';
 
 export interface MetadataProps {
```
Comment (wgwz) [inline web-registry/src/components/molecules/ProjectPageMetadata/ProjectPageMetadata.tsx:13]: microscopic detail; i don't think this needs to exported

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.utils.ts [hunk 19]
```diff
@@ -35,25 +41,86 @@ export const getDisplayAdmin = (address?: string): User | undefined => {
...
-    };
-  }
+type AnchoredMetadata = {
+  projectName?: string;
+  area?: string;
+  areaUnit?: string;
+  placeName?: string;
+};
+
+export const getAnchoredMetadata = (
```
Comment (wgwz) [inline web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.utils.ts:51]: nit; this name felt misleading to me. when i read "get" i think that we are needing to make request somewhere. to me this functions task seems like more of a "parse". so maybe `parseAnchoredMetadata` is an improvement?

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.utils.ts [hunk 20]
```diff
@@ -35,25 +41,86 @@ export const getDisplayAdmin = (address?: string): User | undefined => {
...
+  videoURL?: string | null;
+  glanceText?: string[];
+  primaryDescription?: string;
+  quote?: ProjectQuote;
+  landStewardPhoto?: string | null;
+  landStewardStoryTitle?: string | null;
+  landStewardStory?: string | null;
+};
+
+export const getUnanchoredMetadata = (
```
Comment (wgwz) [inline web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.utils.ts:76]: similar nit here; in both of these cases we already "got" the metadata, and now we are parsing it but no very strong opinion here.

File: web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx [hunk 21]
```diff
@@ -119,46 +121,37 @@ function ProjectDetails(): JSX.Element {
...
-  // Legacy projects use project.metadata. On-chain projects use IRI resolver.
-  const projectTableMetadata: ProjectMetadataLD = project?.metadata;
-  const iriResolvedMetadata = useQuery(
-    getMetadataQuery({ iri: onChainProject?.metadata }),
-  );
-  const metadata = iriResolvedMetadata.data ?? projectTableMetadata;
-
-  const managementActions =
-    metadata?.['regen:landManagementActions']?.['@list'];
+  /** Anchored project metadata comes from IRI resolver. */
```
Comment (wgwz) [inline web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx:124]: another nit; might be worth reviewing and updating some of the other comments in this file. as i have had a little trouble understanding some of them.

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx [hunk 22]
```diff
@@ -58,70 +73,68 @@ function ProjectTopSection({
...
           <Box sx={{ pt: { xs: 5, sm: 6 } }}>
             <ProjectPlaceInfo
               iconClassName={classes.icon}
-              // TODO Format and show on-chain project location if no off-chain location
-              place={
-                metadata?.['schema:location']?.['place_name'] ||
-                metadata?.['schema:location']?.['geojson:place_name'] ||
-                onChainProject?.jurisdiction
-              }
+              // TODO Format on-chain jurisdiction if no anchored location
```
Comment (wgwz) [inline web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:136]: is this todo still needed?

**Review by wgwz — APPROVED** [review]
utACK

File: web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx (hunk 21 above)
Comment (blushi) [reply web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx:124]: which ones exactly?

Comment (wgwz) [reply web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx:124]: it was this block of code and the comments with it, that threw me off a little bit:

```
  // if projectId is handle, query project by handle
  const { data: projectByHandle, isInitialLoading: loadingProjectByHandle } =
    useQuery(
      getProjectByHandleQuery({
        client: graphqlClient,
        enabled: !!projectId && !isOnChainId,
        handle: projectId as string,
      }),
    );

  const projectByHandleOnChainId =
    projectByHandle?.data.projectByHandle?.onChainId ?? undefined;
  const onChainProjectId = isOnChainId ? projectId : projectByHandleOnChainId;
  // else fetch project by onChainId
  const {
    data: projectByOnChainId,
    isInitialLoading: loadingProjectByOnChainId,
  } = useQuery(
    getProjectByOnChainIdQuery({
      client: graphqlClient,
      enabled: !!onChainProjectId,
      onChainId: onChainProjectId as string,
    }),
  );
```

it mentions if/else logic, and there is no explicit if/else statement. after looking at the code long enough i could see why there is an if/else logic here via the `enabled` parameter. it is sort of hidden and just confused me a bit. my preference would be to have an explicit if/else block. but no strong opinions, it makes sense after looking at it for a bit

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx [hunk 23]
```diff
@@ -38,14 +41,21 @@ import {
...
-  getDisplayDeveloper,
+  isAnchoredProjectMetadata,
+  parseOffChainProject,
+  parseProjectMetadata,
+  parseProjectPageMetadata,
 } from './ProjectTopSection.utils';
 
 function ProjectTopSection({
-  data,
+  offChainProject,
```
Comment (wgwz) [inline web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:51]: 👍 appreciate this set of changes, i think it reads better this way.

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx [hunk 24]
```diff
@@ -58,70 +68,59 @@ function ProjectTopSection({
...
-  const landStewardStory = metadata?.['regen:landStewardStory'];
+  const {
+    videoURL,
+    glanceText,
+    primaryDescription,
+    quote,
+    landStewardPhoto,
+    landStewardStoryTitle,
+    landStewardStory,
+  } = parseProjectPageMetadata(projectPageMetadata);
```
Comment (wgwz) [inline web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:85]: 👍 nice!

File: web-registry/src/lib/rdf/rdf.compacted-context.ts [hunk 25]
```diff
@@ -1,31 +1,57 @@
 import * as jsonld from 'jsonld';
 import { JsonLdDocument, NodeObject } from 'jsonld';
 
+import { ANCHORED_PROJECT_CONTEXT, UNANCHORED_PROJECT_CONTEXT } from './rdf';
+
 export const COMPACTED_CONTEXT = {
```
Comment (wgwz) [inline web-registry/src/lib/rdf/rdf.compacted-context.ts:6]: i feel like a comment describing a little bit about what this is used for could be helpful. i.e. maybe something like:

> this context is meant to be used on any incoming JSON-LD document.
> we use this as a way to normalize the format of any incoming JSON-LD document.
> it simplifies parsing these documents in our application.

**Review by wgwz — APPROVED** [review]
utACK, i'm happy with the latest changes, thanks for those @blushi . this is looking pretty good.

could you explain why we have some graphql changes here?

File: web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx (hunk 21 above)
Comment (blushi) [reply web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx:124]: you cannot use react hooks within if/else conditions, it's part of the rules: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level (and [reasons behind it](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)), hence the use of `enabled` which is already largely used across the app for handling such queries

File: web-registry/src/lib/rdf/rdf.compacted-context.ts (hunk 25 above)
Comment (blushi) [reply web-registry/src/lib/rdf/rdf.compacted-context.ts:10]: good idea, done: https://github.com/regen-network/regen-web/pull/1701/commits/97b20a023085d5e1046baa71f1314ba464b82867

**Comment by blushi** [comment]
> could you explain why we have some graphql changes here? (just for my own better understanding)

On the project page, we use two graphql queries conditionally based on the `projectId` route params which can be either a project db handle: `ProjectByHandle` (for querying legacy projects like https://app.regen.network/project/wilmot) or a project on chain id: `ProjectByOnChainId` (eg https://app.regen.network/project/C01-002). So I've just changed those 2 graphql queries to use the same underlying graphql fragment on `Project` so we're sure we always get the same type returned `ProjectFieldsFragment` for this data that we can use in helper functions. This also cleans up those graphql queries since most fields were just copied/pasted.

File: web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx (hunk 22 above)
Comment (blushi) [reply web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:122]: I think so, it'd be a nice to have

Comment (blushi) [reply web-registry/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:122]: although we do require the project metadata to have a location...

File: web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx (hunk 21 above)
Comment (wgwz) [reply web-registry/src/components/templates/ProjectDetails/ProjectDetails.tsx:124]: thanks for explaining it :-)

**Comment by blushi** [comment]
@Mergifyio backport redwood

**Comment by blushi** [comment]
@Mergifyio backport redwood
